packages/flame/lib/src/components/mixins/has_paint.dart

```diff
@@ -3,19 +3,47 @@ import 'dart:ui';
 import 'package:flame/components.dart';
 import 'package:flame/src/palette.dart';
 
-/// Adds a collection of paints to a component
+/// Adds a collection of paints and paint layers to a component
 ///
 /// Component will always have a main Paint that can be accessed
 /// by the [paint] attribute and other paints can be manipulated/accessed
 /// using [getPaint], [setPaint] and [deletePaint] by a paintId of generic type
-/// [T], that can be omitted if the component only have one paint.
+/// [T], that can be omitted if the component only has one paint.
+/// [paintLayers] paints should be drawn in list order during the render. The
+/// main Paint is the first element.
 mixin HasPaint<T extends Object> on Component {
-  final Map<T, Paint> _paints = {};
+  Map<T, Paint>? _paintsCollection;
+  Map<T, Paint> get _paints => _paintsCollection ?? (_paintsCollection = {});
```

Comment:
```suggestion
  late Map<T, Paint>? _paints = {};
```

Reply:
Thanks, I didn't know `late` could do that too.